backend/worker/pe_scripts/sync_hibp_pe.py

```diff
@@ -0,0 +1,210 @@
+    ON CONFLICT (breach_name) 
+    DO UPDATE SET modified_date = EXCLUDED.modified_date,
+    exposed_cred_count = EXCLUDED.exposed_cred_count,
+    password_included = EXCLUDED.password_included;""" % (
+        table,
+        ",".join(columns),
+    )
+    values = [[value for value in dict.values()] for dict in jsonList]
+    cursor = conn.cursor()
+    try:
+        extras.execute_values(cursor, sql, values)
+        conn.commit()
+        print("Data inserted using execute_values() successfully..")
+    except (Exception, psycopg2.DatabaseError) as err:
+        print(err)
+        cursor.close()
+
+
+def execute_hibp_emails_values(conn, jsonList, table):
+    "SQL 'INSERT' of a datafame"
+    columns = jsonList[0].keys()
+    sql = """INSERT INTO %s(%s) VALUES %%s 
+    ON CONFLICT (email, breach_name) 
+    DO NOTHING;""" % (
+        table,
+        ",".join(columns),
+    )
+    values = [[value for value in dict.values()] for dict in jsonList]
+    cursor = conn.cursor()
+    # try:
+    extras.execute_values(cursor, sql, values)
+    conn.commit()
+    print("Data inserted using execute_values() successfully..")
+    # except (Exception, psycopg2.DatabaseError) as err:
+    #     show_psycopg2_exception(err)
+    #     cursor.close()
+
+
+def getDataSource(conn, source):
+    cur = conn.cursor()
+    sql = f"""SELECT * FROM data_source WHERE name='{source}'"""
+    cur.execute(sql)
+    source = cur.fetchone()
+    cur.close()
+    return source
+
+
+try:
+    # Connect to PE DB
+    try:
+        PE_conn = connect(DB_HOST, PE_DB_NAME, PE_DB_USERNAME, PE_DB_PASSWORD)
+        print("Connected to PE database.")
+    except:
+        print("Failed To Connect to PE database")
+
+    # Query PE Db to get the Organization UID
+    try:
+        print(f"Running on organization: {org_name}")
+        cur = PE_conn.cursor()
+        sql = f"""SELECT organizations_uid FROM organizations WHERE name='{org_name}'"""
+        cur.execute(sql)
+        pe_org_uid = cur.fetchone()[0]
+        cur.close()
+        print(f"PE_org_uid: {pe_org_uid}")
+    except:
+        print("Failed with Select Statement")
+        print(traceback.format_exc())
+
+    # Get the Hibp data source uid
+    try:
+        source_uid = getDataSource(PE_conn, "HaveIBeenPwnd")[0]
+        print("Success fetching the data source")
+    except:
+        print("Failed fetching the data source.")
+
+    # Connect to Crossfeed DB
+    try:
+        CF_conn = connect(DB_HOST, CF_DB_NAME, CF_DB_USERNAME, CF_DB_PASSWORD)
+        print("Connected to crossfeed database.")
+    except:
+        print("Failed To Connect to crossfeed database")
+
+    try:
+        # Get a list of all HIBP Vulns for this organization
+        sql = f"""SELECT vuln."structuredData", dom."fromRootDomain", dom."name"
+                FROM domain as dom
+                JOIN vulnerability as vuln 
+                ON vuln."domainId" = dom.id 
+                WHERE dom."organizationId" ='{org_id}' 
```

Comment:
So this query is done for each P&E organization. So I guess the only way to have complete seperation would be to query all the P&E orgs in the JS layer loop through each of them and then call the python script for each org. It would require a bit of a script overhaul but I think it is doable.

Reply:
Also is there a way to pass an object from JS to python?

Reply:
https://github.com/cisagov/crossfeed/pull/1413/commits/ef524dd88346b57fe53cde5d987d9ae6196b9c3e